docs/features/wait/walk.md

```diff
@@ -0,0 +1,12 @@
+# Walk
```

Comment:
docs: we'll need to add it to the mkdocs.yaml file too. See https://github.com/testcontainers/testcontainers-go/pull/2896#discussion_r1858517337